app/models/user_context.rb

```diff
@@ -0,0 +1,12 @@
+class UserContext < DelegateClass(User)
+  attr_reader :acting_as_organization_id
+
+  def initialize(user, organization_id)
+    @acting_as_organization_id = organization_id
+    super(user)
+  end
+
+  def acting_as_organization
+    Organization.find(acting_as_organization_id)
```

Comment:
Would this fail if an action is submitted without an acting organization (i.e. a user doesn't belong to an organization)?

Reply:
Indeed it would. Actually I don't think it needs to exist really - I added it as a convenience method and then didn't actually end up using it.